### PR TITLE
[Intel MKL] enable depthwise convolution backward side code

### PR DESCRIPTION
**Pull request by guizili0**

### CODE REVIEW

File: tensorflow/core/kernels/mkl_conv_grad_filter_ops.cc [hunk 1]
```diff
@@ -485,13 +485,27 @@ class MklConvCustomBackpropFilterOp
...
+          TensorShape diff_filter_tf_shape(
+              {bwd_output_dims[MklDnnDims::Dim_H],
+               bwd_output_dims[MklDnnDims::Dim_W],
+               bwd_output_dims[MklDnnDims::Dim_I],
+               bwd_output_dims[MklDnnDims::Dim_O]});
+          AllocateOutputSetMklShape(context, 0, &diff_filter_tensor,
+                                    diff_filter_tf_shape,
+                                    diff_filter_mkl_shape);
+        } else {
+          // Depthwise Conv2d: bwd_output_dims is GOIHW format
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_conv_grad_filter_ops.cc]: It's not clear to me how `GOIHW` maps to `HWGO` below. Could you please explain more in the comment? Thank you!

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for your PR and sorry for my delay!

File: tensorflow/core/graph/mkl_layout_pass.cc [hunk 2]
```diff
@@ -259,6 +259,9 @@ class MklLayoutRewritePass : public GraphOptimizationPass {
     csinfo_.conv3d_grad_input = "Conv3DBackpropInputV2";
     csinfo_.conv3d_grad_filter = "Conv3DBackpropFilterV2";
     csinfo_.depthwise_conv2d = "DepthwiseConv2dNative";
+    csinfo_.depthwise_conv2d_grad_input = "DepthwiseConv2dNativeBackpropInput";
```
[the remark this answers is not in the document]
Comment (guizili0) [reply tensorflow/core/graph/mkl_layout_pass.cc]: @penpornk This name is consistent with the original Eigen OP name, so we leave the 'd' not 'D'.
the Eigen OP register in https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/ops/nn_ops.cc#L495

File: tensorflow/core/kernels/mkl_conv_grad_filter_ops.cc (hunk 1 above)
Comment (guizili0) [reply tensorflow/core/kernels/mkl_conv_grad_filter_ops.cc]: | TensorFlow       | MKLDNN
--------------------------------------------------------------------
filter_out_depth | depth_multiplier | depth_multiplier * group_count
filter_in_depth  | in_depth         | in_depth / group_count
For depthwise convolution, we have group_count == in_depth.

So here G = original I, and I = 1.
And the GOIHW is mkldnn format, here is try to extract the TF format, TF format is HWIO, as G = original I, so here is HWGO.

File: tensorflow/core/graph/mkl_layout_pass.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (penpornk) [reply tensorflow/core/graph/mkl_layout_pass.cc]: Got it. Let's leave it this way then. Sorry for not checking before!

File: tensorflow/core/kernels/mkl_conv_grad_filter_ops.cc (hunk 1 above)
Comment (penpornk) [reply tensorflow/core/kernels/mkl_conv_grad_filter_ops.cc]: Understood. Thank you! Could you please also add the explanation as comment in the code?

File: tensorflow/core/kernels/mkl_conv_grad_filter_ops.cc [hunk 3]
```diff
@@ -485,13 +485,37 @@ class MklConvCustomBackpropFilterOp
...
+          //                  | TensorFlow       | MKLDNN
+          // ----------------------------------------------------------------
+          // filter_out_depth | depth_multiplier | depth_multiplier *
+          //                  |                  | group_count
+          // ----------------------------------------------------------------
+          // filter_in_depth  | in_depth         | in_depth / group_count
+          // For depthwise convolution, we have group_count == in_depth.
+          // So here G = original I, and I = 1.
+          // And the GOIHW is mkldnn format, here is try to extract the TF
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_conv_grad_filter_ops.cc]:
```suggestion
          // And the GOIHW is mkldnn format, here we try to extract the TF
```

**Review by penpornk — CHANGES_REQUESTED** [review]
One last minor typo fix, please!

File: tensorflow/core/kernels/mkl_conv_grad_filter_ops.cc (hunk 1 above)
Comment (penpornk) [reply tensorflow/core/kernels/mkl_conv_grad_filter_ops.cc]: Thank you very much!

**Review by penpornk — APPROVED** [review]
Thank you very much!

**Comment by penpornk** [comment]
I agree they are not related. Tagging this ready-to-pull now.

**Comment by guizili0** [comment]
@penpornk thank you very much!

**Pull request merged**